### ISSUE TITLE
Edit symbol when not a recognized contributor greyed out with sometimes-wrong hover message

### ISSUE THREAD

**Issue by wolftune** [go-gitea/gitea#20881]
### Description

Reproduce the bug:

- Visit a project
- Fork it
- Go back to the original project
- Open a file
- Hover over the pencil symbol to edit

Expected behavior:

- Editing should work but should switch you to your fork (or prompt you to switch to the fork at least)

Experienced behavior:

- Action is greyed out
- Hover text says "You must fork this repository to make or propose changes to this file." even though you already did fork the repository

### Screenshots

![fork-hover-message](https://user-images.githubusercontent.com/2801853/185769946-a57af18d-1602-42ba-a7ed-0f5a5cdb64f1.png)

### Gitea Version

1.17

### Can you reproduce the bug on the Gitea demo site?

Yes

### Operating System

Ubuntu GNU/Linux

### Browser Version

Firefox 103.0

**Comment by zeripath** [comment]
Did you refresh the page after you forked the repo?

**Comment by wolftune** [comment]
> Did you refresh the page after you forked the repo?

Yes, at both Codeberg and at https://try.gitea.io, and with multiple attempts. The behavior is consistent. Anyway, when you first fork, you land at your fork, and you have to go back to the original before you can even be in the situation to see this bug. So, refreshing shouldn't matter, but the issue shows up even after forcing a reload.

**Comment by bircni** [comment]
This looks resolved by #34240 (Edit file workflow for creating a fork and proposing changes), which will ship in 1.25.0. The edit/delete buttons are no longer greyed out when you lack write access to the branch — clicking edit now takes you through a fork-based workflow, and when a usable fork already exists it edits on a branch in your fork instead of showing the misleading "you must fork" message. Closing as fixed; please reopen if you still hit this on 1.25.
